runtime/int.go

```diff
@@ -66,6 +66,14 @@ func (i *Int) IsTrue() bool {
 // IntType is the object representing the Python 'int' type.
 var IntType = newBasisType("int", reflect.TypeOf(Int{}), toIntUnsafe, ObjectType)
 
+func intAbs(f *Frame, o *Object) (*Object, *BaseException) {
+	z := toIntUnsafe(o).Value()
+	if z <= 0 {
```

Comment:
To avoid an unnecessary allocation in the positive case, how about:

```
z := toIntUnsafe(o)
if z.Value() > 0 {
  return z.ToObject(), nil
}
// .. check for overflow as mentioned below
return NewInt(-x).ToObject(), nil
```